library-model/library-model-generator-integration-test/src/test/java/com/uber/nullaway/libmodel/LibraryModelIntegrationTest.java

```diff
@@ -0,0 +1,76 @@
+package com.uber.nullaway.libmodel;
+
+import com.google.errorprone.CompilationTestHelper;
+import com.uber.nullaway.NullAway;
+import java.util.Arrays;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class LibraryModelIntegrationTest {
+
+  @Rule public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  private CompilationTestHelper compilationHelper;
+
+  @Before
+  public void setup() {
+    compilationHelper = CompilationTestHelper.newInstance(NullAway.class, getClass());
+  }
+
+  @Test
+  public void libraryModelNullableReturnsTest() {
+    compilationHelper
+        .setArgs(
+            Arrays.asList(
+                "-d",
+                temporaryFolder.getRoot().getAbsolutePath(),
+                "-XepOpt:NullAway:AnnotatedPackages=com.uber",
+                "-XepOpt:NullAway:JarInferEnabled=true",
+                "-XepOpt:NullAway:JarInferUseReturnAnnotations=true"))
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import com.uber.nullaway.libmodel.AnnotationExample;",
+            "class Test {",
+            "  static AnnotationExample annotationExample = new AnnotationExample();",
+            "  static void test(String value){",
+            "  }",
+            "  static void testPositive() {",
+            "    // BUG: Diagnostic contains: passing @Nullable parameter 'annotationExample.makeUpperCase(\"nullaway\")'",
+            "    test(annotationExample.makeUpperCase(\"nullaway\"));",
+            "  }",
+            "  static void testNegative() {",
+            "    test(annotationExample.nullReturn());",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void libraryModelNullableReturnsArrayTest() {
+    compilationHelper
+        .setArgs(
+            Arrays.asList(
+                "-d",
+                temporaryFolder.getRoot().getAbsolutePath(),
+                "-XepOpt:NullAway:AnnotatedPackages=com.uber",
+                "-XepOpt:NullAway:JarInferEnabled=true",
+                "-XepOpt:NullAway:JarInferUseReturnAnnotations=true"))
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import com.uber.nullaway.libmodel.AnnotationExample;",
+            "class Test {",
+            "  static AnnotationExample annotationExample = new AnnotationExample();",
+            "  static void test(Integer[] value){",
+            "  }",
+            "  static void testPositive() {",
+            "    // BUG: Diagnostic contains: passing @Nullable parameter 'annotationExample.generateIntArray(7)'",
+            "    test(annotationExample.generateIntArray(7));",
```

Comment:
Fun! This wasn't possible in my version of the `astubx` format, correct? 🙂 

Edit: Ah, never mind, it uses the same position of annotation as the case above, we haven't introduced yet a way to annotate both the elements of an array and the array itself, correct? But this parses the annotation in the correct/JSpecify location.

Reply:
Yes, you are correct.  jspecify/jdk puts the annotation in the right spot, so if you parse it the incorrect way you get incorrect specs :-)  You're right that we don't yet support annotations in both positions.  Do you think this needs a code comment @lazaroclapp?

Reply:
Wouldn't hurt, but I also think that behavior will get less and less surprising with time as the "JSpecify way" becomes the default overall for NullAway and arrays (at least for type-use annotations).